verify.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+
+CLI="faas-cli"
+func_name="validate"
+
+build_template() 
+{
+    template=$1
+
+    echo Building $template
+    $CLI new $func_name --lang $template 2>/dev/null 1>&2
+    $CLI build -f $func_name.yml
+}
+
+verify_and_clean()
+{
+    image=$1
+    tag_name=latest
+    
+    echo Verifying $template
+    container=$(docker run -d -p 8080:8080 $func_name:$tag_name)
+    sleep 5
+    output=$(curl -s -d "testing" http://127.0.0.1:8080)
+    
+    echo $image output: $output
+    success=false
+    if [ ! -z "$output" ]
+    then # output was not empty = good template
+        success=true
+    fi
+
+    echo Cleaning $image
+    docker rm $container -f 2>/dev/null 1>&2    
+    docker rmi $func_name:$tag_name 2>/dev/null 1>&2
+
+    # delete the generated function folder/file
+    rm -rf validate validate.yml 
+
+    if [ "$success" = false ]
+    then
+        echo $image template failed validation
+        exit 1
+    else
+        echo $image template validation successful
+    fi
+}
+
+if ! [ -x "$(command -v faas-cli)" ]; then
+    HERE=`pwd`
+    cd /tmp/
+    curl -sL https://github.com/openfaas/faas-cli/releases/download/0.6.9/faas-cli > faas-cli
```

Comment:
It would be nice if the version was a variable

Reply:
It should use the installation script for Linux. Could also be a pre-install step for Travis